fix: Improve NextGen metadata migration field mapping

elvisomondi opened the pull request:
- Fix NextGen alt text mapping to FooGallery caption field
- Fix NextGen description mapping to FooGallery description field
- Add image slug/filename mapping for proper WordPress titles
- Update Plugin base class to handle slug field properly

Fixes metadata preservation issues during NextGen to FooGallery migration. Before: NextGen descriptions incorrectly went to captions, alt text was lost After: Proper field mapping preserves all metadata in correct FooGallery fields

elvisomondi committed:
fix: Improve NextGen metadata migration field mapping

- Fix NextGen alt text mapping to FooGallery caption field
- Fix NextGen description mapping to FooGallery description field
- Add image slug/filename mapping for proper WordPress titles
- Leave alt text field empty as requested
- Update Plugin base class to handle slug field properly

Fixes metadata preservation issues during NextGen to FooGallery migration.
Before: NextGen descriptions incorrectly went to captions, alt text was lost
After: Proper field mapping preserves all metadata in correct FooGallery fields

bradvin commented on includes/plugins/class-nextgen.php:
```diff
@@ -102,8 +102,10 @@ private function find_images( $gallery_id, $gallery_path ) {
+                    'slug' => $nextgen_image->filename,
+                    'caption' => $nextgen_image->alttext,
+                    'description' => $nextgen_image->description,
+                    'alt' => '',
```
just checking this is right? We are always setting the alt to blank?

elvisomondi replied:
NextGen has two options, the "Alt & Title Text" section and the "Description" section as shown here: https://prnt.sc/sYnXJbIK2OTf
The changes made ensure that the "Alt & Title Text" matches with the "Caption" in FooGallery and the "Description" matches with the "Description" field in FooGallery as shown here: https://prnt.sc/EaXVdWu2IZR1
This was not happening before and those fields were always blank

bradvin commented on includes/plugins/class-nextgen.php:
```diff
@@ -102,8 +102,10 @@ private function find_images( $gallery_id, $gallery_path ) {
-                    'caption' => $nextgen_image->description,
-                    'alt' => $nextgen_image->alttext,
+                    'slug' => $nextgen_image->filename,
+                    'caption' => $nextgen_image->alttext,
```
Just checking this logic - we want to map the nextgen alt text to the attachment caption?

elvisomondi committed:
fix: Correct NextGen title field mapping to use alt text instead of filename

- Fix NextGen alt text mapping to properly populate FooGallery title field
- Add missing title property to Image class constructor
- Update attachment creation to use title instead of slug for post_title
- Ensure NextGen alt text appears as image title in FooGallery, not filename
- Maintain proper field mapping: NextGen alttext -> FooGallery title & alt fields

Resolves issue where migrated images showed filename as title instead of NextGen alt text.

elvisomondi committed:
Complete NextGen metadata migration improvements
- Fix NextGen alt text mapping to properly populate FooGallery title and alt fields
- Configure FooGallery galleries to display image titles as caption titles

elvisomondi committed:
- Update plugin version to 1.4 with improved metadata handling
  - Update readme with changelog for NextGen metadata improvements
